lib/ace/editor.js

```diff
@@ -74,7 +75,13 @@ var Editor = function(renderer, session) {
     this.renderer = renderer;
 
     this.commands = new CommandManager(useragent.isMac ? "mac" : "win", defaultCommands);
-    this.textInput  = new TextInput(renderer.getTextAreaContainer(), this);
+    
+    var iOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream;
+    if (iOS) {
+      this.textInput = new TextInput_iOS(renderer.getTextAreaContainer(), this);
```

Comment:
@etamponi re comment above, could you get away with appending a class to the editor here? and using the css to target based on that classname rather than viewport size?